src/client/pythonEnvironments/common/commonUtils.ts

```diff
@@ -29,10 +32,10 @@ export async function* findInterpretersInDir(
     recurseLevel?: number,
     filterSubDir?: FileFilterFunc,
     ignoreErrors = true,
-): AsyncIterableIterator<string> {
+): AsyncIterableIterator<DirEntry> {
     // "checkBin" is a local variable rather than global
-    // so we can stub it out during unit testing.
-    const checkBin = getOSType() === OSType.Windows ? isWindowsPythonExe : isPosixPythonBinPattern;
+    // so we can stub out getOSType() during unit testing.
+    const checkBin = getOSType() === OSType.Windows ? windows.matchPythonExeFilename : posix.matchPythonExeFilename;
```

Comment:
For posix `matchPythonExeFilename` sounds weird. 
```suggestion
    const checkBin = getOSType() === OSType.Windows ? windows.matchPythonExeFilename : posix.matchPythonBinFilename;
```

Reply:
I used "Exe" because there is no reason they have to be a binary.  That said, I don't mind changing it to "bin".